src/rotation_generator.jl

```diff
@@ -246,3 +246,30 @@ function Base.show(io::IO, ::MIME"text/plain", X::RotationGenerator)
     io = IOContext(io, :typeinfo => eltype(X))
     Base.print_array(io, X)
 end
+
+"""
+    isrotationgenerator(r)
+    isrotationgenerator(r, tol)
+
+Check whether `r` is a rotation generator matrix (skew-symmetric matrix).
+"""
+isrotationgenerator
+
+function isrotationgenerator(r::StaticMatrix{N,N,T}) where {N,T<:Real}
+    m = SMatrix(r)
+    return iszero(m+m')
+end
+
+function isrotationgenerator(r::AbstractMatrix{T}) where T<:Real
+    if !=(size(r)...)
```

Comment:
To reduce subtlety and confusion

```suggestion
    if size(r,1) != size(r, 2)
```